components/studio/bin/hab-studio.sh

```diff
@@ -1076,6 +1076,35 @@ chown_artifacts() {
   fi
 }
 
+# **Internal**  Run a command which may fail without aborting whole script
+try() {
+  local status
+  if [[ $- == *e* ]]; then
+    # Disable `set -e` as briefly as possible to avoid masking other errors
+    set +e
+    $*
+    status=$?
+    set -e
+  else
+    $*
+    status=$?
+  fi
+
+  if [ $status -ne 0 ]; then
+    echo "Warning: '$*' failed with status $status"
+  fi
+}
+
+# **Internal** Unmount mount point if mounted
+# Don't abort script on umount failure
+# ARGS: [umount_options] <mount_point>
+try_umount() {
+  eval _mount_point=\$$# # getting the last arg is surprisingly hard
+  if $bb mount | $bb grep "on $_mount_point type"; then
```

Comment:
I think you'd still want `grep -q` to suppress the grep matches if any are found.